src/map.cpp

```diff
@@ -5645,6 +5636,50 @@ void map::remove_field( const tripoint &p, const field_id field_to_remove )
     }
 }
 
+void map::add_splatter( const field_id type, const tripoint &where, int intensity )
```

Comment:
It prevents removing fields using negative `intensity` and IMHO more consistent (in case some other changes to _splatters_ will be made).